pkg/crc/machine/kubeconfig.go

```diff
@@ -264,19 +272,45 @@ func mergeConfigHelper(kubeConfigFile, globalConfigFile string) error {
 	if err != nil {
 		return err
 	}
+	// append cluster name to the AuthInfos
+	cfg, err := appendClusterToAuthinfos(currentConf)
+	if err != nil {
+		return err
+	}
 	// Merge the currentConf to globalConfig
-	for name, cluster := range currentConf.Clusters {
+	for name, cluster := range cfg.Clusters {
 		globalConf.Clusters[name] = cluster
 	}
 
-	for name, authInfo := range currentConf.AuthInfos {
+	for name, authInfo := range cfg.AuthInfos {
 		globalConf.AuthInfos[name] = authInfo
 	}
 
-	for name, context := range currentConf.Contexts {
+	for name, context := range cfg.Contexts {
 		globalConf.Contexts[name] = context
 	}
 
-	globalConf.CurrentContext = currentConf.CurrentContext
+	globalConf.CurrentContext = cfg.CurrentContext
 	return clientcmd.WriteToFile(*globalConf, globalConfigPath)
 }
+
+func appendClusterToAuthinfos(cfg *api.Config) (*api.Config, error) {
+	var clusterURL string
+	for _, cluster := range cfg.Clusters {
+		url, err := url.Parse(cluster.Server)
+		if err != nil {
+			return cfg, err
+		}
+		clusterURL = url.Host
+	}
+	for name, authInfo := range cfg.AuthInfos {
+		delete(cfg.AuthInfos, name)
+		username := fmt.Sprintf("%s/%s", name, clusterURL)
```

Comment:
This is essentially a duplication:

```
clusterUser := fmt.Sprintf("%s/%s", username, url.Host)
```

Why is the previous logic not re-used?

Reply:
Something like?

```golang
func appendClusternameToUser(username, url) string

	url, err := url.Parse(clusterConfig.ClusterAPI)
	if err != nil {
		return err
	}
	return fmt.Sprintf("%s/%s", username, url.Host)
}
```